ddl/table.go

```diff
@@ -208,6 +208,32 @@ func onTruncateTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		job.State = model.JobStateCancelled
 		return ver, errors.Trace(err)
 	}
+
+	var oldPartitionIDs []int64
+	if tblInfo.GetPartitionInfo() != nil {
+		oldPartitionIDs = getPartitionIDs(tblInfo)
+		newPi := &model.PartitionInfo{
+			Type:    tblInfo.Partition.Type,
+			Expr:    tblInfo.Partition.Expr,
+			Columns: tblInfo.Partition.Columns,
+			Enable:  tblInfo.Partition.Enable,
+		}
+		for _, def := range tblInfo.Partition.Definitions {
+			pid, err := genGlobalID(d)
+			if err != nil {
+				job.State = model.JobStateCancelled
+				return ver, errors.Trace(err)
+			}
+			piDef := model.PartitionDefinition{
```

Comment:
Cloud we just reset `pid`？ Then we needn't create `PartitionInfo` and `PartitionDefinition`.